src/main/java/org/mockito/internal/junit/ArgMismatchFinder.java

```diff
@@ -0,0 +1,28 @@
+package org.mockito.internal.junit;
+
+import org.mockito.internal.invocation.finder.AllInvocationsFinder;
+import org.mockito.invocation.Invocation;
+import org.mockito.internal.invocation.Stubbing;
+
+/**
+ * For given mocks, finds stubbing arg mismatches
+ */
+class ArgMismatchFinder {
```

Comment:
No subclassing. I hate violently inheritance :D (love interfaces, though).

I prefer objects than static methods (as we are already discussing elsewhere ;)